src/uwtools/tests/test_rocoto.py

```diff
@@ -0,0 +1,72 @@
+# pylint: disable=missing-function-docstring
+"""
+Tests for uwtools.rocoto module.
+"""
+from importlib import resources
+
+import yaml
+
+from uwtools import rocoto
+from uwtools.tests import support
+
+# Test functions
+
+
+def test_add_jobname():
+    expected = yaml.safe_load(
+        """
+task_hello:
+  command: echo hello 
+  jobname: hello
+metatask_howdy:
+  var:
+    mem: 1 2 3
+  task_howdy_#mem#:
+    command: echo hello 
+    jobname: howdy_#mem#
+  task_hola_#mem#:
+    command: echo hello 
+    jobname: hola_#mem#
+  metatask_hey_#mem#:
+    var:
+      day: mon tues
+    task_hey_#mem#_#day#:
+      command: echo hello 
+      jobname: hey_#mem#_#day#
+"""
```

Comment:
Honest question: Is this block of YAML more than we actually need to test the `add_jobname()` functionality? It's not a problem to use incomplete, unrealistic bits of data in tests; in fact, having extraneous data that is not actually used by the code-under-test can distract from understanding the test. I'd probably recommend removing unneeded data here if it's not needed for the test.

Reply:
I think making testing the functionality of the `_add_jobname` can be made smaller such as:

`def test_add_jobname():
    expected = yaml.safe_load(
        """
task_hello:
  command: echo hello 
  jobname: hello
metatask_howdy:
  var:
    mem: 1 2 3
  task_howdy_#mem#:
    command: echo hello 
    jobname: howdy_#mem#` 

Thoughts? @christinaholtNOAA and @maddenp-noaa

Reply:
Here's the rub. Each of these `task_` sections needs to be a dictionary on input to `_add_jobname`, otherwise the logic breaks down. We can certainly remove the `var:` entries under metatasks without a problem.

I'm also curious if it makes sense to require the input type to be a dict of dicts...

Reply:
I missed the last part that Naureen asked. Do we want to simplify the input dict, or make sure that several tasks and nested metatasks are tested, or is the abridged version above enough?

Reply:
I think we need at least one task and one metatask, as otherwise the recursive step won't be tested. I don't think we necessarily need more than one of each, though.

Reply:
I simplified the expected output and task example down further, removing `var` altogether.

Reply:
It might be good to keep one non-task element in the tree, even something like `foo: bar`, to demonstrate that nothing goes wrong if a non-task element is present. Assuming that that's something we should expect to happen in practice. @christinaholtNOAA ?